pandas/core/frame.py

```diff
@@ -9460,7 +9495,12 @@ def to_period(self, freq=None, axis: Axis = 0, copy: bool = True) -> DataFrame:
         setattr(new_obj, axis_name, new_ax)
         return new_obj
 
-    def isin(self, values) -> DataFrame:
+    def isin(
+        self,
+        values: Union[
```

Comment:
leave this off for now. i am not sure this is right and this is complicated.

Reply:
Done